### ISSUE TITLE
False positive with collection initializers `IDE0028`

### ISSUE THREAD

**Issue by niquenen** [dotnet/roslyn#70854]
**Version Used**: .NET `8.0.100`

**Steps to Reproduce**:

> [!NOTE]
> macOS 14.1.1 (23B81, arm64) used for this test.

1. Create an empty project.

   ```sh
   dotnet new console -o issue
   ```
2. Create a `.globalconfig` file at the root of the project.
3. Put the following configuration into `.globalconfig`.

   ```ini
   # Disable all rules
   dotnet_analyzer_diagnostic.category-Design.severity = none
   dotnet_analyzer_diagnostic.category-Documentation.severity = none
   dotnet_analyzer_diagnostic.category-Globalization.severity = none
   dotnet_analyzer_diagnostic.category-Interoperability.severity = none
   dotnet_analyzer_diagnostic.category-Maintainability.severity = none
   dotnet_analyzer_diagnostic.category-Naming.severity = none
   dotnet_analyzer_diagnostic.category-Performance.severity = none
   dotnet_analyzer_diagnostic.category-SingleFile.severity = none
   dotnet_analyzer_diagnostic.category-Reliability.severity = none
   dotnet_analyzer_diagnostic.category-Security.severity = none
   dotnet_analyzer_diagnostic.category-Style.severity = none
   dotnet_analyzer_diagnostic.category-Usage.severity = none

   # IDE0028: Collection initialization can be simplified
   dotnet_diagnostic.IDE0028.severity = warning
   dotnet_style_collection_initializer = true
   ```
4. Open `issue.csproj` and add a `PropertyGroup`.

   ```xml
   <Project Sdk="Microsoft.NET.Sdk">

     <PropertyGroup>
       <OutputType>Exe</OutputType>
       <TargetFramework>net8.0</TargetFramework>
       <ImplicitUsings>enable</ImplicitUsings>
       <Nullable>enable</Nullable>
     </PropertyGroup>

    <!-- Enable .NET analyzers -->
     <PropertyGroup>
       <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
       <EnableNETAnalyzers>true</EnableNETAnalyzers>
     </PropertyGroup>

   </Project>
   ```
5. Open `Program.cs` and use the following example.

   ```cs
   // Collection initialization can be simplified csharp(IDE0028)
   List<int> list01 = new List<int>();
   //                 ~~~

   // No warning
   IList<int> list02 = new List<int>();

   // Collection initialization can be simplified csharp(IDE0028)
   List<int> list03 = new List<int>{};
   //                 ~~~

   // No warning
   IList<int> list04 = new List<int>{};
   ```

   [Syntax tree on sharplab](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABADAAhQRgCwG4BYAKBJQGYsAmDAYQwG8SMWtKVcMBZACgEpGzVsID0IugHsANlICmAY2ABLCQDsMS1UuVgpSgF5hlajPLDqARrIwQlAWwAOegGZLZsDAG0lECAFdZaA1fAIBdeQgACzAAJwceAEkAEQBRNDQqAA4+IWEWABkfYAAeTWAAPgw9KDRsDABeDFVZAHcMQqhS1Qr+ImI81jEB4eEAP3GSXLyhgDkJDBbYrVUAcynhBI6Sssrq4AyGptb2oq6evj71wfFaaTlFFXVNbSVdAyNH03MMKxt7JyUrncXh8/kCwTB4SisXiyTSGWyVwKpx2VSKaEojWabS2Z3KDAAvn0RkMRiNxqNJv1hrN5osYss1tSBpsUd1dujOFjjridoTiawCSQhcQgA==).
6. Run `dotnet run`.

   ```console
   foo:~$ dotnet run

   Program.cs(2,20): warning IDE0028: Collection initialization can be simplified (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0028) [issue.csproj]
   Program.cs(9,20): warning IDE0028: Collection initialization can be simplified (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0028) [issue.csproj]
   ```

**Diagnostic Id**:

`IDE0028`: [Use collection initializers](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0028).

**Expected Behavior**:

The analyzer should not report a warning.

**Actual Behavior**:

Could be a false positive, a warning is thrown.

**Comment by CyrusNajmabadi** [comment]
Three are expected. The collection initialization can be simplified.  In this case, in those cases it can be simplified to `[]`.
